xbmc/filesystem/FileCache.cpp

```diff
@@ -277,13 +277,13 @@ void CFileCache::Process()
 
     while (m_writeRate)
     {
-      if (m_writePos - m_readPos < m_writeRate)
+      if (m_writePos - m_readPos < m_writeRate * g_advancedSettings.m_readBufferFactor)
```

Comment:
In principle the way I've changed it now, shouldn't affect this behavior. It should only prevent the above issue I just mentioned. And to me it makes more sense to have this logic contained in CFileCache rather than in CDVDPlayer.